terraform/sql-database-server/main.tf

```diff
@@ -1,6 +1,6 @@
 # https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/sql_server
 
-resource "azurerm_sql_server" "adl_sql" {
+resource "azurerm_mssql_server" "adl_sql" {
```

Comment:
⚠️ **[tfsec]** <[AVD-AZU-0026](https://aquasecurity.github.io/tfsec/v1.28.1/checks/azure/database/secure-tls-policy/)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Database server does not require a secure TLS version.

Reply:
Created a minimum_tls_version whose default value is the highest supported version of TLS, and explicitly set the minimum_tls_version property of the azurerm_msssql_server resource to such variable.